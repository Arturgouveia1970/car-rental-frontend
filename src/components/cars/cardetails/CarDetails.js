// eslint-plugin-react-hooks
import React, { useEffect, useState } from 'react';
import { NavLink, useParams } from 'react-router-dom';
import axios from 'axios';
import styles from './CarDetails.module.css';
// import Sidebar from '../../common/sidebar/Sidebar';

const CarDetails = () => {
  const { id } = useParams();
  const [car, setCar] = useState({});
  const [loading, setLoading] = useState(false);

  // fetch data callback function vvv asyncronous
  // const fetchData = useCallback(async (id) => {
  //   setLoading(true);
  //   // await axios get function
  //   await axios
  //     .get(`https://dreamcars2.onrender.com/api/v1/car/${id}`)
  //     .then((response) => {
  //       setCar(response.data);
  //       // console.log(response.data);
  //       setLoading(false);
  //     });
  // }, []);
  // useEffect(() => {
  //   // call the callback function
  //   fetchData(id);

  //   // include the function in the dependencies
  // }, [id, fetchData]);

  // // const newLocal = <Sidebar />;
  // if (loading) {
  //   return (
  // <div className="container-fluid vh-100 v-100 d-flex justify-content-center align-items-center">
  //       <i className="fa-solid fa-spinner fa-spin fs-1" />
  //     </div>
  //   );
  // }

  useEffect(() => {
    axios
      .get(`https://dreamcars2.onrender.com/api/v1/car/${id}`)
      .then((response) => setCar(response.data.car));
    setLoading(false);
  }, [id]);

  if (loading) {
    return (
      <div className="container-fluid vh-100 v-100 d-flex justify-content-center align-items-center">
        <i className="fa-solid fa-spinner fa-spin fs-1" />
      </div>
    );
  }

  return (
    <>
      {/* {newLocal} */}
      <h2 className={styles.title}>CAR DETAILS</h2>
      <div className={styles.detailContainer}>
        <div className={styles.car}>
          <img src={car.image} alt={car.model} />
        </div>
        <div className={styles.detail}>
          <div className={styles.carModel}>
            <h2>{car.model}</h2>
          </div>
          <div className={styles.carYear}>
            {/* <p>year:</p> */}
            <p>{car.year}</p>
          </div>
          <div className={styles.price}>
            <p className={styles.priceColor}>
              $
              {car.price}
            </p>
            <p>pd </p>
          </div>
          <NavLink to={`/reserve/${id}`} className={styles.btncontainer}>
            <button type="button" className={styles.reservebtn}>
              <i className="fa-solid fa-car-side me-4" />
              Reserve
            </button>
          </NavLink>
        </div>
      </div>
    </>
  );
};
export default CarDetails;
