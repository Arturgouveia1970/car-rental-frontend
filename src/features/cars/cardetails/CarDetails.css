h1 {
  color: rgb(59, 163, 33);
}
