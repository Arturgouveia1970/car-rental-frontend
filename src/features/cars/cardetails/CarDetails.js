import React from 'react';
import './CarDetails.css';
import Sidebar from '../../common/sidebar/Sidebar';
// import CarsListHeader from '../carlist/CarsListHeader';
import car1 from '../../../assets/images/Honda-Car.jpg';

const CarDetails = () => (
  <div className="carslist-cont">
    <Sidebar />
    <div className="car-details">
      <img src={car1} alt="car" />
      <div className="text">
        <h4>HONDA CX 2010</h4>
        <h4>$250.00 p/d</h4>
        <button type="button" className="book-btn">Reserve now</button>
      </div>
    </div>
  </div>
);

export default CarDetails;
