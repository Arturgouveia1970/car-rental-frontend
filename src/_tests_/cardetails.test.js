import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CarDetails from '../features/cars/cardetails/CarDetails';
import store from '../Redux/configureStore';

describe('Car detail tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <CarDetails />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
