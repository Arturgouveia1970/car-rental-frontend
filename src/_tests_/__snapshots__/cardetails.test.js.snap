// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car detail tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="title"
      >
        CAR DETAILS
      </h2>
      <div
        class="detailContainer"
      >
        <div
          class="car"
        >
          <img />
        </div>
        <div
          class="detail"
        >
          <div
            class="carModel"
          >
            <h2 />
          </div>
          <div
            class="carYear"
          >
            <p />
          </div>
          <div
            class="price"
          >
            <p
              class="priceColor"
            >
              $
            </p>
            <p>
              pd 
            </p>
          </div>
          <a
            class="btncontainer"
            href="/reserve/undefined"
          >
            <button
              class="reservebtn"
              type="button"
            >
              <i
                class="fa-solid fa-car-side me-4"
              />
              Reserve
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h2
      class="title"
    >
      CAR DETAILS
    </h2>
    <div
      class="detailContainer"
    >
      <div
        class="car"
      >
        <img />
      </div>
      <div
        class="detail"
      >
        <div
          class="carModel"
        >
          <h2 />
        </div>
        <div
          class="carYear"
        >
          <p />
        </div>
        <div
          class="price"
        >
          <p
            class="priceColor"
          >
            $
          </p>
          <p>
            pd 
          </p>
        </div>
        <a
          class="btncontainer"
          href="/reserve/undefined"
        >
          <button
            class="reservebtn"
            type="button"
          >
            <i
              class="fa-solid fa-car-side me-4"
            />
            Reserve
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
